### PR TITLE
Propagate network info for containers sharing network namespaces

### LINKED ISSUE #1394: Containers using other containers net namespace don't have IP addresses
This particularly affects k8s, where every container uses a pause containers net namespace.

The IP should show up in the details panel, and we should ignore pause containers for the purpose of rendering.

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #1394

### CODE REVIEW

**Comment by paulbellamy** [comment]
I'm curious why the difference from `containerWithHostIPsRenderer`? It would be nice to do both the same way (either way is fine).

**Comment by tomwilkie** [comment]
Yeah I was thinking of pull the `--net:host` stuff into here as well.  Will do.

**Comment by paulbellamy** [comment]
Have also realized this will not handle `--net=container:containerName`

**Comment by tomwilkie** [comment]
> Have also realized this will not handle --net=container:containerName

As discuss, this is a misfeature that we've decided not to support.

File: probe/docker/reporter.go [hunk 1]
```diff
@@ -96,66 +147,73 @@ func (r *Reporter) Report() (report.Report, error) {
...
+		hostNetworkInfo := report.EmptySets
+		if hostIPs, err := getLocalIPs(); err == nil {
+			hostIPsWithScopes := addScopeToIPs(r.hostID, hostIPs)
+			hostNetworkInfo = hostNetworkInfo.
+				Add(ContainerIPs, report.MakeStringSet(hostIPs...)).
+				Add(ContainerIPsWithScopes, report.MakeStringSet(hostIPsWithScopes...))
+		}
+
+		var NetworkInfo func(prefix string) report.Sets
+		NetworkInfo = func(prefix string) report.Sets {
```
Comment (paulbellamy) [inline probe/docker/reporter.go]: Why is this capital (and defined on 2 lines?)

**Comment by paulbellamy** [comment]
Quibble about naming, but otherwise, LGTM.

**Pull request merged**